### DISCUSSION TITLE
det_box_type='poly'直接传这个参数会报错，但是我又需要用，求解决

### DISCUSSION THREAD

**Discussion by lance000001** [PaddlePaddle/PaddleOCR#13002] [Q&A]
故事是这样的
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/83149ccd-63f6-485f-8db4-0e662ec38e64)
我项目里的这些斜着的字有几率识别不出来
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/92861f27-9cfd-493e-97e1-d6a5e821ef4b)
就是自己finetune过，没传参的推理
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/8c509bc4-65ef-4ce4-840d-429fa7218949)
于是我追踪代码到了这里，并强制在这里把det_box_type参数改成了'poly'
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/b3c4457a-2c0f-4965-aea8-c455c4ca4ae7)
很神奇的就识别出来了
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/ad7e2a7c-2cad-494e-9f9a-8b947538dc54)
于是我尝试传参使用
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/31125043/62ad9e61-c7d9-4e20-a3ab-60e99dc96c24)
结果报错，能修复嘛

**Comment by 07571026** [comment]
我也遇到一樣的報錯，錯誤原因是list裡面含有兩組以上的array時，將list變成轉成np.array時，裡面的array的維度必須相同，否則就會出錯
因為dt_boxes_new裡面有兩組以上的array，並且他們的shape[0]不同，所以出錯

我調整的方式你參考一下
根據你的錯誤是在 predict_det.py 的 line 261
det_boxes = np.array(dt_boxes_new)
請把這行備註掉，並且把下面的 return det_boxes 備註起來
改成 return det_boxes_new
實際修改如下圖
<img width="338" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/47208945/fb2ed474-74a6-43b8-8e91-45eea3b71752">
